Sources/Common/Background Queue/Queue.swift

```diff
@@ -48,42 +49,29 @@ public protocol Queue: AutoMockable {
         // sourcery:TypeCast="AnyEncodable(data)"
         data: TaskData,
         groupStart: QueueTaskGroup?,
-        blockingGroups: [QueueTaskGroup]?
-    ) -> ModifyQueueResult
+        blockingGroups: [QueueTaskGroup]?,
+        onComplete: @escaping (ModifyQueueResult) -> Void
+    )
 
     func run(onComplete: @escaping () -> Void)
 
     func deleteExpiredTasks()
 }
 
 public extension Queue {
-    func addTask<TaskData: Codable>(
-        type: String,
-        // sourcery:Type=AnyEncodable
-        // sourcery:TypeCast="AnyEncodable(data)"
-        data: TaskData
-    ) -> ModifyQueueResult {
-        addTask(type: type, data: data, groupStart: nil, blockingGroups: nil)
-    }
-
-    func addTask<TaskData: Codable>(
-        type: String,
-        // sourcery:Type=AnyEncodable
-        // sourcery:TypeCast="AnyEncodable(data)"
-        data: TaskData,
-        groupStart: QueueTaskGroup?
-    ) -> ModifyQueueResult {
-        addTask(type: type, data: data, groupStart: groupStart, blockingGroups: nil)
-    }
-
+    // convenient addTask alternative that doesn't require you give a value for all parameters
     func addTask<TaskData: Codable>(
         type: String,
         // sourcery:Type=AnyEncodable
         // sourcery:TypeCast="AnyEncodable(data)"
         data: TaskData,
-        blockingGroups: [QueueTaskGroup]?
-    ) -> ModifyQueueResult {
-        addTask(type: type, data: data, groupStart: nil, blockingGroups: blockingGroups)
```

Comment:
The same purpose is satisfied when we had 3 separate methods and now we have 1. We are trying to add convenience to ourselves so when we call `backgroundQueue.addTask()`, we don't need to provide all of the parameters if the caller doesn't need to provide a value for a paremeter. 

3 were provided originally using the method overloading technique. But while I was working on this PR, I looked at the code and had an idea to instead of using method overloading, instead provide default values for all optional parameters. This refactor allowed us to simplify the code and go from 3 to 1 method while staying backwards compatible with all callers in our SDK code.